lib/cachex/warmer.ex

```diff
@@ -67,19 +67,37 @@ defmodule Cachex.Warmer do
       # enforce the behaviour
       @behaviour Cachex.Warmer
 
+      def warm(pid, opts) do
+        if opts[:respect_async_warmers] do
+          GenServer.call(pid, :cachex_warmer, :infinity)
+        else
+          send(pid, :cachex_warmer)
+        end
+      end
+
       @doc false
       # Initializes the warmer from a provided state.
       #
       # Initialization will trigger an initial cache warming, and store
       # the provided state for later to provide during further warming.
-      def init({cache, warmer(async: async, state: state)}) do
+      def init({cache, warmer}) do
+        {:ok, {cache, warmer, nil}}
+      end
+
+      def handle_call(
+            :cachex_warmer,
+            _from,
+            {cache, warmer(async: async) = warmer, timer_ref}
+          ) do
+        if timer_ref, do: Process.cancel_timer(timer_ref)
```

Comment:
This is an unrelated fix - but currently every time that `Cachex.warm` is called we would stack timers causing additional runs of the warmer, by tracking and potentially canceling a timer, we should just reset the `interval` whenever `Cachex.warm`

Reply:
Good catch on this one, I'm going to revisit this after merge - I can clean up the `Cachex.warm/2` implementation a bit more.